js/runner.go

```diff
@@ -318,9 +321,50 @@ func (r *Runner) SetOptions(opts lib.Options) error {
 		r.console = c
 	}
 
+	if opts.DNS != nil {
+		// FIXME: Resolver probably shouldn't be reset here...
+		// It's done because the js.Runner is created before the full
+		// configuration has been processed, at which point we don't have
+		// access to the DNSConfig, and need to wait for this SetOptions
+		// call that happens after all config has been assembled.
+		// We could make DNSConfig part of RuntimeOptions, but that seems
+		// conceptually wrong since the JS runtime doesn't care about it
+		// (it needs the actual resolver, not the config), and it would
+		// require an additional field on Bundle to pass the config through,
+		// which is arguably worse than this.
+		ttl, err := parseTTL(opts.DNS.TTL.String)
+		if err != nil {
+			return err
+		}
+		r.Resolver = netext.NewDNSResolver(ttl, opts.DNS.Strategy.DNSStrategy)
+	}
+
 	return nil
 }
 
+func parseTTL(ttlS string) (types.NullDuration, error) {
+	ttl := types.NewNullDuration(0, false)
+	switch ttlS {
+	case "", "inf":
+		// cache indefinitely
+		ttl.Valid = true
+	case "0":
+		// disable cache
+	default:
+		origTTLs := ttlS
+		// Treat unitless values as seconds
```

Comment:
Given https://github.com/loadimpact/k6/issues/1305, I'd prefer to have be milliseconds, not seconds. Even though it makes less sense for this particular context, the real UX benefit would be to treat duration configurations the same everywhere in k6, we shouldn't introduce any more special cases

Reply:
Hhmm I agree with not introducing special cases, but in this case treating them as milliseconds would be counter-intuitive, especially since we don't log what the interpreted value is (maybe we should?). But OK, will change it.

Reply:
Resolved in the latest changes.